portal-impl/src/com/liferay/portal/util/HttpImpl.java

```diff
@@ -1262,21 +1262,17 @@ public String shortenURL(String url, int count) {
 					sb.append(qName);
 					sb.append(StringPool.EQUAL);
 					sb.append(newURL);
-
-					if (i < (params.length - 1)) {
-						sb.append(StringPool.AMPERSAND);
-					}
+					sb.append(CharPool.AMPERSAND);
 				}
 			}
 			else {
 				sb.append(param);
-
-				if (i < (params.length - 1)) {
-					sb.append(StringPool.AMPERSAND);
-				}
+				sb.append(CharPool.AMPERSAND);
 			}
 		}
 
+		sb.setStringAt(StringPool.BLANK, sb.index() - 1);
```

Comment:
If you want to throw away the last element, you should do sb.setIndex(sb.index() - 1), don't replace the last element to blank string, that is very wasteful.